react/components/ViewOrder/OrderButtons.js

```diff
@@ -17,19 +26,34 @@ const OrderActions = ({ order, allowSAC }) => {
     )
   }
 
-  const showEditOrderButton = allowSAC && allowEdition
+  useEffect(() => {
+    async function fetchLoggedInEmail() {
+      const customerEmail = await getCustomerEmail()
+
+      setLoggedInEmail(customerEmail)
+    }
```

Comment:
is there a reason for this function being inside a useEffect?

Reply:
That was the only way I could figure out to get async/await working 🤔

Reply:
This example should work:

```ts
  const fetchLoggedInEmail =  async () => {
      try {
         const customerEmail = await getCustomerEmail()
         setLoggedInEmail(customerEmail)
     } catch (error) {
         console.error(error)
         return null;
     }
    }
  useEffect(() => {
    fetchLoggedInEmail().then(() => {
       // success
    })
    return () => {
      setLoggedInEmail('')
    }
  }, [setLoggedInEmail]);
```

Behind the scenes, await/async works exactly as a Promise, it's just a way to avoid nested Promises.

Reply:
Now I see what you meant. Originally I thought you were asking why I used `useEffect` at all. 

At first I tried this:
```
useEffect(() => {
  const email = await getCustomerEmail()
  ...
```
but that gave me a message saying that I couldn't use `await` directly inside `useEffect`. It suggested that I structure my code the way I had it before (with the function nested inside `useEffect`) but your suggestion is better. I just committed the change.